src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/publicApi/SubjectAccessRequestService.kt

```diff
@@ -19,28 +21,34 @@ class SubjectAccessRequestService(
   private val licenceEventRepository: LicenceEventRepository,
   private val auditEventRepository: AuditEventRepository,
 ) {
+
   @Transactional
   fun getSarRecordsById(nomisId: String): SarContent? {
-    val licences =
-      licenceRepository.findAllByNomsId(nomisId)
+    val licences = licenceRepository.findAllByNomsId(nomisId)
+
     if (licences.isEmpty()) return null
+
     val licenceIds = licences.map { it.id }
+    val auditEvents = getAuditEvents(licenceIds)
+    val licenceEvents = getLicenceEvents(licenceIds)
+    val sarLicences = licences.map { transformToSarLicence(licenceService.getLicenceById(it.id)) }
```

Comment:
do we we need to do the separate call? 
It'd be nicer to transform the entity directly - maybe as a follow up you could investigate getting rid of this n+1?